src/scripts/main.js

```diff
@@ -1,3 +1,257 @@
 'use strict';
 
-// write your code here
+/// /// /// /// /// /// ///
+//    APP LOGIC
+/// /// /// /// /// /// ///
+
+let score = 0;
+let isPlaying = false;
+
+const row1 = [0, 1, 2, 3];
+const row2 = [4, 5, 6, 7];
+const row3 = [8, 9, 10, 11];
+const row4 = [12, 13, 14, 15];
+
+const column1 = [0, 4, 8, 12];
+const column2 = [1, 5, 9, 13];
+const column3 = [2, 6, 10, 14];
+const column4 = [3, 7, 11, 15];
+
+const rightDirection = [column3, column2, column1].flat();
+const leftDirection = [column2, column3, column4].flat();
+const upDirection = [row2, row3, row4].flat();
+const downDirection = [row3, row2, row1].flat();
+
+/// /// /// /// /// /// ///
+/// ELEMENTS SELECTION AND CRIATION
+/// /// /// /// /// /// ///
+
+const cells = document.querySelectorAll('.field-cell');
+const startButton = document.querySelector('.start');
+const restartButton = document.createElement('button');
+const messageLoseElement = document.querySelector('.message-lose');
+const messageWinElement = document.querySelector('.message-win');
+const messageStartElement = document.querySelector('.message-start');
+
+restartButton.classList.add('button', 'restart');
+restartButton.textContent = 'Restart';
+restartButton.hidden = true;
+document.querySelector('.controls').append(restartButton);
+
+/// /// /// /// /// /// ///
+/// HELPER FUNCTIONS
+/// /// /// /// /// /// ///
+const checkIfHaveMoves = () => {
+  if ([...cells].filter(cell => cell.classList.length === 1).length > 0) {
+    return true;
+  }
+
+  let haveMoves = false;
+
+  cells.forEach((cell, i) => {
+    const topCell = row1.includes(i) ? null : cells[i - 4];
+    const bottomCell = row4.includes(i) ? null : cells[i + 4];
+    const leftCell = column1.includes(i) ? null : cells[i - 1];
+    const rightCell = column4.includes(i) ? null : cells[i + 1];
+
+    const adjacentCells = [topCell, bottomCell, leftCell, rightCell];
+
+    adjacentCells.forEach(adjacentCell => {
+      if (!adjacentCell) {
+        return;
+      }
+
+      if (adjacentCell.textContent === cell.textContent) {
+        haveMoves = true;
```

Comment:
add `return` here; if we know that we have moves, we may not check other elements